dns/dns.go

```diff
@@ -27,50 +31,51 @@ type DNSCollector struct {
 
 	hosts []string
 
-	count      map[string]float64
-	errorCount map[string]float64
-
-	countMutex      sync.Mutex
-	errorCountMutex sync.Mutex
+	latencyHistogramVec  *histogramvec.HistogramVec
+	latencyHistogramDesc *prometheus.Desc
 
-	total      *prometheus.Desc
-	errorTotal *prometheus.Desc
-	latency    *prometheus.Desc
+	errorTotal     float64
+	errorTotalDesc *prometheus.Desc
 }
 
 func NewCollector(config Config) (*DNSCollector, error) {
 	if config.Logger == nil {
-		return nil, microerror.Maskf(exporterkit.InvalidConfigError, "%T.Logger must not be empty", config)
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
```

Comment:
I'm being a bit cheeky and sliding in my error fixes here too, please forgive me <3